tests/graph/ahead_behind.c

```diff
@@ -0,0 +1,51 @@
+#include "clar_libgit2.h"
+
+static git_repository *_repo;
+static git_commit *commit;
+static size_t ahead;
+static size_t behind;
+
+void test_graph_ahead_behind__initialize(void)
+{
+    git_oid oid;
+    cl_git_pass(git_repository_open(&_repo, cl_fixture("testrepo.git")));
+
+    git_oid_fromstr(&oid, "be3563ae3f795b2b4353bcce3a527ad0a4f7f644");
+    cl_git_pass(git_commit_lookup(&commit, _repo, &oid));
+}
+
+void test_graph_ahead_behind__cleanup(void)
+{
+	git_commit_free(commit);
+	commit = NULL;
+
+	git_repository_free(_repo);
+	_repo = NULL;
+}
+
+void test_graph_ahead_behind__returns_correct_result(void)
+{
+    
+    git_oid oid;
+    git_oid oid2;
+    git_commit *other;
+
+    git_oid_fromstr(&oid, "e90810b8df3e80c413d903f631643c716887138d");
+    git_oid_fromstr(&oid2, "be3563ae3f795b2b4353bcce3a527ad0a4f7f644");
+    
+    cl_assert_equal_i(0, git_graph_ahead_behind(&ahead, &behind, _repo, &oid, &oid2));
+
+    cl_assert_equal_i(0, git_graph_ahead_behind(&ahead, &behind, _repo, git_commit_id(commit), git_commit_id(commit)));
+    cl_git_pass(git_commit_nth_gen_ancestor(&other, commit, 1));
+
+    cl_assert_equal_i(0, git_graph_ahead_behind(&ahead, &behind, _repo, git_commit_id(commit), git_commit_id(other)));
+    cl_assert_equal_i(0, git_graph_ahead_behind(&ahead, &behind, _repo, git_commit_id(other), git_commit_id(commit)));
+
+    git_commit_free(other);
+
+    cl_git_pass(git_commit_nth_gen_ancestor(&other, commit, 3));
+
+    cl_assert_equal_i(0, git_graph_ahead_behind(&ahead, &behind, _repo, git_commit_id(commit), git_commit_id(other)));
+    cl_assert_equal_i(0, git_graph_ahead_behind(&ahead, &behind, _repo, git_commit_id(other), git_commit_id(commit)));
```

Comment:
You have several tests that ensure that `git_graph_ahead_behind` _works_, but you're not actually examining the results of `ahead` and `behind` to ensure that we're returning the correct results.

You should be able to look at the test resource directory in question (I tend to copy it to a temp folder, then rename `.gitted` to `.git`.  Then run the ahead/behind counts with git itself, and use that to test `ahead` and `behind`.  (eg, `cl_assert_equal_sz(42, ahead)`...